tests/checkpoint/test_download.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import tempfile
+from unittest.mock import patch
+
+import pytest
+import torch
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.distributed.fsdp import StateDictType
+
+from composer.checkpoint import download_monolithic_checkpoint
+from composer.utils import dist
+from tests.checkpoint.helpers import init_model
+from tests.common.markers import world_size
+from tests.utils.test_remote_uploader import DummyObjectStore
+
+
+@world_size(1, 2)
+@pytest.mark.gpu
+@pytest.mark.parametrize('rank_zero_only', [True, False])
+def test_download_monolithic_checkpoint(world_size: int, rank_zero_only: bool):
+    # Write a checkpoint
+    tmp_dir = tempfile.TemporaryDirectory()
+    use_fsdp = False
+    if world_size > 1:
+        use_fsdp = True
+    fsdp_model, _ = init_model(use_fsdp=use_fsdp)
+
+    if use_fsdp:
+        with FSDP.state_dict_type(fsdp_model, StateDictType.FULL_STATE_DICT):
+            state = fsdp_model.state_dict()
+    else:
+        state = fsdp_model.state_dict()
```

Comment:
you can also use `get_model_state_dict` function
```suggestion
    from torch.distributed.checkpoint.state_dict import get_model_state_dict, StateDictOptions
    get_model_state_dict(model, options=StateDictOptions(full_state_dict=True))
```